### PR TITLE
Feature: sdk instruction buy tests

### PR DESCRIPTION
**Pull request by tbiedukhin**
Jira - https://metaplx.atlassian.net/browse/MET-8

### Notes
* Introduced test for buy method

### CODE REVIEW

File: membership-token/js/test/buy.test.ts [hunk 1]
```diff
@@ -0,0 +1,203 @@
...
+    initSellingResourceTx,
+    [sellingResource],
+    defaultSendOptions,
+  );
+
+  logDebug('buy-sol:: created selling resource', sellingResource.publicKey.toBase58());
+  addLabel('create:selling-resource', sellingResource.publicKey.toBase58());
+  assertConfirmedTransaction(t, initSellingResourceRes.txConfirmed);
+
+  const treasuryMint = new PublicKey('11111111111111111111111111111111');
```
Comment (kurpav) [inline membership-token/js/test/buy.test.ts:111]: why there is the string?
I remember there is a const smth like `NATIVE_MINT`

File: membership-token/js/test/create-store.test.ts [hunk 2]
```diff
@@ -36,47 +38,3 @@ test('create-store: success', async (t) => {
 
   assertConfirmedTransaction(t, createStoreRes.txConfirmed);
 });
-
-test('create-store: bad name', async (t) => {
```
Comment (kurpav) [inline membership-token/js/test/create-store.test.ts:40]: why is it removed?

File: membership-token/js/src/accounts/Store.ts [hunk 3]
```diff
@@ -102,17 +107,17 @@ export class StoreAccountData {
...
     accountDiscriminator: number[];
   }
 >(
   [
-    ['accountDiscriminator', beet.fixedSizeArray(beet.u8, 8)],
+    ['accountDiscriminator', beet.uniformFixedSizeArray(beet.u8, 8)],
     ['admin', beetSolana.publicKey],
-    ['name', beet.fixedSizeUtf8String(NAME_MAX_LEN)],
-    ['description', beet.fixedSizeUtf8String(DESCRIPTION_MAX_LEN)],
+    ['name', beet.utf8String],
```
Comment (zaxozhu) [inline membership-token/js/src/accounts/Store.ts:119]: It looks like having this you are able to get rid of checkByteSizes in [createStore.ts](https://github.com/metaplex-foundation/metaplex-program-library/blob/master/membership-token/js/src/instructions/createStore.ts) and [createMarket.ts](https://github.com/metaplex-foundation/metaplex-program-library/blob/master/membership-token/js/src/instructions/createMarket.ts) 

```typescript
  const name = checkByteSizes(args['name'], NAME_MAX_LEN);
  const description = checkByteSizes(args['description'], DESCRIPTION_MAX_LEN);
```

Comment (tbiedukhin) [reply membership-token/js/src/accounts/Store.ts:119]: type has been changed after api generation, should we keep fixed size manually or we able to move forward with utf8string type

File: membership-token/js/test/create-store.test.ts (hunk 2 above)
Comment (tbiedukhin) [reply membership-token/js/test/create-store.test.ts:40]: https://github.com/metaplex-foundation/metaplex-program-library/pull/151#discussion_r792420237

File: membership-token/js/src/accounts/Store.ts (hunk 3 above)
Comment (zaxozhu) [reply membership-token/js/src/accounts/Store.ts:119]: Please check it by adding one more test for createStore.ts with a short store name & description. If the local validator allows it, then it's fine.

Comment (tbiedukhin) [reply membership-token/js/src/accounts/Store.ts:119]: added

**Review by zaxozhu — DISMISSED** [review]

**Review by kurpav — DISMISSED** [review]

**Review by zaxozhu — APPROVED** [review]

**Pull request merged**